### PR TITLE
fix: [M3-8424] - Fix CodeQL alerts for `DOM text reinterpreted as HTML`

### PR DESCRIPTION
**Pull request by carrillo-erik**
## Description 📝
This PR attempts to reproduce and resolve the CodeQL code scan alerts for `DOM text reinterpreted as HTML`. Alternatively, these alerts might be determined to be false positives and will be dismissed as so.

## Changes  🔄
List any change relevant to the reviewer.
- The changes proposed by this PR were generated by GitHub's Copilot Autofix feature.

## Target release date 🗓️
10/16/2024

### Prerequisites
(How to setup test environment)
- Install the CodeQL extension on VSCode
- Install the CodeQL CLI tool using [this guide](https://luisrangelc.medium.com/analyze-vulnerabilities-in-your-code-with-codeql-locally-3e5d31f8ed2)
  - Once you reached the **Create a database of your code with CodeQL** step, `cd` into the `src/` directory of the `manager` package and execute the `codeql database create codeqldb --language=javascript` command.
  - This is the last step you'll follow on this guide.
- In a separate VSCode window, open the `codeql-repo` (this is one of the directories created using the linked guide).
- Click on the CodeQL icon found in the VSCode sidebar.
- Select the DataBase created from executing the `codeql database create codeqldb --language=javascript` command. (This DataBase will be located in the `src/` directory of the `manager` package.

### Verification steps
(How to verify changes)
- In the **Queries** tab, navigate to the `javascript/ql/src/Security/` directory.
- Under **CWE-079** locate the `XssThroughDom.ql` query and run it by clicking the play button.
  - Verify that running the query returns 0 results.
- Under **CWE-116** run all the queries by clicking the play button at the directory level. 
  - Verify that running the query returns 0 results.

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/store/authentication/authentication.requests.ts
Comment (hkhalil-akamai) [inline packages/manager/src/store/authentication/authentication.requests.ts:1]: Was the vulnerability detected in this file a false alarm?

**Review by hkhalil-akamai — APPROVED** [review]
I may have my local CodeQL configured incorrectly but I'm not seeing any results when scanning the selected vulnerabilities either before or after this change. Still, reviewing the changes, I believe they are sensical and will cause no harm.

**Review by coliu-akamai — APPROVED** [review]
In the same boat as Hussain - `encodeURIComponent` makes sense to me. Will the other alerts be dismissed as false positives?

**Comment by carrillo-erik** [comment]
> In the same boat as Hussain - `encodeURIComponent` makes sense to me. Will the other alerts be dismissed as false positives?

My hope is that by merging this small change, we should see if the alert goes away or at least trigger a new alert, which may provide more context. I will check with John and Jaalah before I take an action with respect to handling the alert on the GitHub page.

**Comment by abailly-akamai** [comment]
@carrillo-erik This PR is being reverted. Not only did it not fix the dependabot warnings but it introduced a bad bug - see https://github.com/linode/manager/pull/11017

**Pull request merged**